mimesis/schema.py

```diff
@@ -63,24 +64,37 @@ def __call__(self, name: Optional[str] = None,
         if name is None:
             raise UndefinedField()
 
-        providers = []
-        if '.' in name:
-            cls, name = name.split('.', 1)
-            providers.append(cls)
-
-        # TODO: Optimization.
-        for provider in providers or dir(self.gen):
-            if hasattr(self.gen, provider):
-                provider = getattr(self.gen, provider)
-                if name in dir(provider):
-                    method = getattr(provider, name)
-                    result = method(**kwargs)
-                    if key and callable(key):
-                        return key(result)
-                    return result
-        else:
-            raise UnsupportedField(name if not providers
-                                   else '{}.{}'.format(providers[0], name))
+        def tail_parser(tails, obj):
```

Comment:
Maybe:
```
def tail_parser(tails: str, obj: Any) -> Any:
    # ...
```

Reply:
@lk-geimfari Right, forgot about it. Thanks! 👍 
Notify me, plz, when you do full review 😃

Reply:
@duckyou That's all. As I can judge everything is excellent. Just annotate this function and let's merge it!